src/core/orders/use-cases/OrderManagement.js

```diff
@@ -87,6 +91,18 @@ class OrderManagement {
     return this.#toOrderDTO(updatedOrder);
   }
 
+  async checkout(orderId) {
+    const orderDTO = await this.orderRepository.findById(orderId);
```

Comment:
**question**: Dessa forma, os métidis dos repositórios não vão ter padrão sobre o formato de resposta. Alguns estão retornando as entities e alguns estão retornando o DTO. Vamos seguir assim? 
Não seria melhor fazer essa conversão dentro do repo?

Reply:
Não entendi a questão...
Se for relacionado ao DTO, a ideia é que os repositórios sempre retornos apenas os objetos e não as próprias entidades, mas a princípio todos seguem o mesmo formato (retornar o DTO quando foi um método `find`.

A instanciação das Entities fica sendo responsabilidade das classes de dentro do domínio (use-cases)

Reply:
Olhando para os dois repositórios, todos os métodos que possuem algum retorno estão chamando o createOrderDTO, então nenhum deles está criando entities...

Reply:
Lugares que estão instanciando as entities:
<img width="455" alt="image" src="https://github.com/user-attachments/assets/1e72efc3-6770-4e15-8db4-efd137e8a0f7">